File: chat_service/lib/chat_service/rabbitmq/publisher.ex
defmodule ChatService.Rabbitmq.Publisher do
  alias AMQP.Basic

  @exchange "chat_exchange"

  def publish_message(queue, message) do
    with channel <- ChatService.Rabbitmq.Connection.channel(),
         :ok <- Basic.publish(channel, @exchange, queue, message) do
      :ok
    else
      error -> {:error, error}
    end
  end
end

File: services/translate_service/rabbitmq/publisher.py
import json
from rabbitmq.connection import RabbitMQConnection

class RabbitMQPublisher:
    def __init__(self):
        self.connection = RabbitMQConnection()
        self.channel = self.connection.create_channel()

    def publish(self, queue, message):
        self.channel.basic_publish(exchange="", routing_key=queue, body=json.dumps(message))
        print(f"Mensagem publicada na fila '{queue}': {message}")
